### PR TITLE
common/Graylog.cc: use std move to avoid copy

### PR DESCRIPTION
**Pull request by stiopaa1**
Signed-off-by: Michal Jarzabek <stiopa@gmail.com>

### CODE REVIEW

File: src/common/Graylog.cc [hunk 1]
```diff
@@ -24,7 +24,7 @@ Graylog::Graylog(const SubsystemMap * const s, std::string logger)
       m_log_dst_valid(false),
       m_hostname(""),
       m_fsid(""),
-      m_logger(logger),
+      m_logger(std::move(logger)),
```
Comment (tchaikov) [inline src/common/Graylog.cc]: logger is not a rvalue *reference*. so this does not help with the performance. and neither do i think it's a hotspot here.

Comment (cbodley) [inline src/common/Graylog.cc]: > logger is not a rvalue _reference_

i'm not sure what you're saying here. `m_logger(logger)` does call the `std::string` copy ctor, and adding `std::move()` causes it to use the move ctor instead

your point remains that this is only called once on initialization, so it won't ever show up in the profiler. but a copy is a copy, and I'd prefer to move when it's appropriate

Comment (tchaikov) [inline src/common/Graylog.cc]: sorry, i was wrong. thought that the type of `logger` should be `std::string&&`.

> but a copy is a copy, and I'd prefer to move when it's appropriate

makes sense to me.

**Comment by tchaikov** [comment]
agree with @cbodley , lgtm also.

**Pull request merged**